containers/Ideas/IdeaCreate.js

```diff
@@ -118,37 +118,38 @@ export const IdeaCreate = props => {
         }
       })
       //error handling is provisional and should probably go somewhere else eventually
-      .then((response) => {
+      .then(response => {
         const {statusCode, data} = response
-        if (statusCode == 201) {
+        if (statusCode === 201) {
           Alert.alert('Your idea was added.', 'Thank you for participating!',  [{ text: 'Ok' }])
           props.navigation.goBack()
-        } else if (editing && statusCode == 200) {
+        } else if (editing && statusCode === 200) {
           Alert.alert('Your idea was updated.', '',  [{ text: 'Ok' }])
           props.navigation.navigate({
             name: 'IdeaDetail',
             params: { idea: data },
             merge: true,
           })
-        } else if (statusCode == 400) {
+        } else if (statusCode === 400) {
+          let errorMsg
           if (data && data.image) {
-            const errorString = data.image.reduce((acc, curr) => {
+            errorMsg = data.image.reduce((acc, curr) => {
               return `${acc}\n${curr}`
             }, 'Please try again.\n')
-            setError(errorString)
-            setSubmitPending(false)
           } else {
-            setError('Required fields are missing.')
-            setSubmitPending(false)
+            errorMsg = ('Required fields are missing.')
           }
-        } else if (response.statusCode == 403) {
-          setError(response.data.detail)
-          setSubmitPending(false)
+          return Promise.reject(errorMsg)
+        } else if (statusCode === 403) {
+          return Promise.reject(data.detail)
         } else {
-          setError('Try again.')
-          setSubmitPending(false)
+          return Promise.reject('Try again.')
         }
       })
+      .catch(error => {
```

Comment:
small nitpick/personal preference: move to line above instead of newline?

Reply:
i think it is more consistent to have it on the same indentation as the `.then` above. Because it's chaining, it would be wrong in my opinion if we do not break it into a new line:

```
fetching()
  .then()
  .then()
  .catch()
```

vs.

```
fetching()
  .then()
    .then()
      .catch()
```

Reply:
fair!